### PR TITLE
e1s1 - add request link screen

### PR DESCRIPTION
**Pull request by mikehalmamoj**
On behalf of Nathan Russell.

### CODE REVIEW

File: server/middleware/setupRequestLink.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function setUpRequestLink(): Router {
+  const router = express.Router()
+  const requestLinksController = new RequestLinkController()
+
+  // TODO This uses the same max session age and secret as for a logged in user. Consider separating these into different env vars.
+  router.use(
+    session({
+      cookie: { maxAge: config.session.expiryMinutes * 60000 },
+      store: new session.MemoryStore(),
```
Comment (mikehalmamoj) [inline server/middleware/setupRequestLink.ts:15]: Can we make this a RedisStore?

File: server/routes/link/RequestLinkController.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response } from 'express'
+import RequestLinkView from './RequestLinkView'
+
+export default class RequestLinkController {
+  async getRequestLinkView(req: Request, res: Response) {
+    const view = new RequestLinkView(req.session?.requestLinkForm || {}, req.flash('errors'))
+
+    res.clearCookie('create_barcode_token').render('pages/link/requestLink', view.renderArgs)
```
Comment (mikehalmamoj) [inline server/routes/link/RequestLinkController.ts:8]: Could maybe use a unit test to check the cookie is cleared?

File: server/middleware/setupRequestLink.ts (hunk 1 above)
Comment (NathanAtClarity) [reply server/middleware/setupRequestLink.ts:15]: Yes, though I will probably need some pointers. Next PR ?

File: server/routes/link/RequestLinkController.ts (hunk 2 above)
Comment (NathanAtClarity) [reply server/routes/link/RequestLinkController.ts:8]: 👍 I've just written an integration test for this as part of the next part of dev - can I commit it in the next PR?

File: server/middleware/setupRequestLink.ts (hunk 1 above)
Comment (mikehalmamoj) [reply server/middleware/setupRequestLink.ts:15]: Yeah next PR.

File: server/routes/link/RequestLinkController.ts (hunk 2 above)
Comment (mikehalmamoj) [reply server/routes/link/RequestLinkController.ts:8]: yes next PR

**Review by NathanAtClarity — APPROVED** [review]

File: server/utils/nunjucksSetup.ts [hunk 3]
```diff
@@ -45,4 +54,16 @@ export default function nunjucksSetup(app: express.Express, path: pathModule.Pla
     const array = fullName.split(' ')
     return `${array[0][0]}. ${array.reverse()[0]}`
   })
+
+  njkEnv.addFilter('findError', (array: Error[], formFieldId: string) => {
+    const item = array?.find(error => error.href === `#${formFieldId}`)
+    if (item) {
```
Comment (yasinmustafa) [inline server/utils/nunjucksSetup.ts:60]: could replace this if condition block into a one liner
`return item ? { text: item.message } : null`

**Review by yasinmustafa — APPROVED** [review]
LGTM - one comment

File: server/utils/nunjucksSetup.ts (hunk 3 above)
Comment (NathanAtClarity) [reply server/utils/nunjucksSetup.ts:60]: 👍

**Pull request merged**